golem/network/transport/tcpnetwork.py

```diff
@@ -269,11 +273,18 @@ class BasicProtocol(SessionProtocol):
        serialization
     """
 
-    def __init__(self):
-        super().__init__()
-        self.opened = False
+    def __init__(self, session_factory, **_kwargs):
+        super().__init__(session_factory)
         self.db = DataBuffer()
         self.spam_protector = SpamProtector()
+        self.machine.add_transition_callback(
+            'connectionLostTransition', 'connected', 'disconnected',
+            'before',
+            lambda reason: (
+                (self.session.dropped() if self.session else True)
+                or True  # always True
+            ),

```

Comment:
At this point state machine should guarantee you that `self.session` exists.
```suggestion
            lambda: self.session.dropped(),
```